### PR TITLE
Removes std::regex in favor of re2

### PR DESCRIPTION
**Pull request by tmccrmck**
- Also makes onnxruntime_hosting a dependency of hosting tests to fix boost include problems

### CODE REVIEW

File: cmake/onnxruntime_hosting.cmake [hunk 1]
```diff
@@ -28,12 +28,6 @@ target_include_directories(${PROJECT_NAME} PRIVATE
 
 target_link_libraries(${PROJECT_NAME} PRIVATE
         ${Boost_LIBRARIES}
-        ${PROVIDERS_MKLDNN}
-        ${MKLML_SHARED_LIB}
-        ${PROVIDERS_CUDA}
-        ${onnxruntime_tvm_libs}
-        ${onnxruntime_libs}
```
Comment (NonStatic2014) [inline cmake/onnxruntime_hosting.cmake:35]: NIT, you may want to keep this one and remove the rest of them.

File: cmake/onnxruntime_unittests.cmake [hunk 2]
```diff
@@ -172,8 +172,6 @@ elseif(HAS_FILESYSTEM_H OR HAS_EXPERIMENTAL_FILESYSTEM_H)
     list(APPEND onnxruntime_test_framework_libs stdc++fs)
 endif()
 
-
-
```
Comment (NonStatic2014) [inline cmake/onnxruntime_unittests.cmake:176]: Remove unnecessary empty lines.

**Review by NonStatic2014 — COMMENTED** [review]
You may want to have more test cases for the url parsing scenario, e.g. empty input, complex input, unsupported protocols, etc.

File: cmake/onnxruntime_unittests.cmake (hunk 2 above)
Comment (tmccrmck) [reply cmake/onnxruntime_unittests.cmake:176]: I removed those lines, not added.

File: cmake/onnxruntime_hosting.cmake (hunk 1 above)
Comment (tmccrmck) [reply cmake/onnxruntime_hosting.cmake:35]: I tried this and ${onnxruntime_libs} doesn't give us any of those libs. In fact, I can't find where it is set.

**Pull request merged**